File: 3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int  size,i=0,j,k,max=0;
        size = s.size();
        for(j = 0;j<size;j++){
            for(k = i;k<j;k++)
                if(s[k]==s[j]){
                    i = k+1;
                    break;
                }
            if(j-i+1 > max)
                max = j-i+1;
        }
        return max;
    }
};

//每次一样之后就需要重新定位，定位的是宠物的那个部分的后面一位

File: 3.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int l = s.length(),index=0,j,max= 0;
        for(int i = 0;i<l;i++){
            for(j = index;j<i;j++){
                if((s.charAt(i))==(s.charAt(j))){
                    index = j+1;
                    break;
                }
            }
            if(j - index + 1 > max)
                max = j - index + 1;
        }
        return max;
    }
}
